lib/fluent/plugin/in_forward.rb

```diff
@@ -379,21 +379,33 @@ def check_and_skip_invalid_event(tag, es, remote_host)
     def add_source_info(es, conn)
       new_es = Fluent::MultiEventStream.new
       if @source_address_key && @source_hostname_key
-        address = conn.remote_addr
-        hostname = conn.remote_host
+        address = conn.remote_addr.dup
+        hostname = conn.remote_host.dup
+        if address.ascii_only?
+          address.force_encoding(Encoding::US_ASCII)
```

Comment:
Just for my curiosity, it should be handled for other encoding such as cp932 or other encoding with other if clauses?
i.e. non ascii encoding cases will not be handled as binary with parquet format?